exe/wallet/http-bridge/Main.hs

```diff
@@ -106,144 +122,186 @@ import qualified Network.Wai.Handler.Warp as Warp
                               Main entry point
 -------------------------------------------------------------------------------}
 
-main :: IO ()
-main = runCli withBackend cliDefinition
-
-withBackend :: Manager -> Environment -> IO ()
-withBackend manager env@Environment {..} = do
-    listen <- parseWalletListen env
-    backendPort <- parseArg $ longOption "backend-port"
-    dbFile <- parseOptionalArg $ longOption "database"
-    stateDir <- parseOptionalArg $ longOption "state-dir"
-    let verbosity = verbosityFromArgs args
-    -- FIXME
-    -- We also need to support the special arg 'local' for the bridge... or not.
-    parseArg (longOption "network") >>= \case
-        Testnet ->
-            runCliCommand env manager
-                (execServe @'Testnet verbosity listen backendPort dbFile)
-                (execLaunch "testnet" verbosity listen backendPort stateDir)
-        Mainnet ->
-            runCliCommand env manager
-                (execServe @'Mainnet verbosity listen backendPort dbFile)
-                (execLaunch "mainnet" verbosity listen backendPort stateDir)
+main :: forall (n :: Network). IO ()
+main = join $ execParser $ cli $ mempty
+    <> cmdLaunch
+    <> cmdServe
+    <> cmdMnemonic
+    <> cmdWallet @(HttpBridge n)
+    <> cmdTransaction @(HttpBridge n)
+    <> cmdAddress @(HttpBridge n)
+    <> cmdVersion
 
 {-------------------------------------------------------------------------------
-                               CLI definitions
--------------------------------------------------------------------------------}
-
-cliDefinition :: Docopt
-cliDefinition = makeCli
-    cliCommands
-    cliOptions
-    cliExamples
-
-cliCommands :: String
-cliCommands = [here|
-  cardano-wallet launch [--network=STRING] [(--port=INT | --random-port)] [--backend-port=INT] [--state-dir=DIR] [(--quiet | --verbose )]
-  cardano-wallet serve  [--network=STRING] [(--port=INT | --random-port)] [--backend-port=INT] [--database=FILE] [(--quiet | --verbose )]
-|]
+                            Command - 'launch'
 
-cliOptions :: String
-cliOptions = [here|
-  --backend-port <INT>      port used for communicating with the HTTP bridge [default: 8080]
-|]
+  cardano-wallet launch
+    [--network=STRING]
+    [(--port=INT | --random-port)]
+    [--node-port=INT]
+    [--state-dir=DIR]
+    [(--quiet | --verbose )]
+-------------------------------------------------------------------------------}
 
-cliExamples :: String
-cliExamples = [here|
-  # Launch and monitor a wallet server and its associated chain producer
-  cardano-wallet launch --network mainnet --random-port --state-dir .state-dir
+data LaunchArgs = LaunchArgs
+    { _network :: Either Local Network
+    , _listen :: Listen
+    , _nodePort :: Port "Node"
+    , _stateDir :: Maybe FilePath
+    , _verbosity :: Verbosity
```

Comment:
Yes and No. We could, but then, we would have to add a pragma to suppress errors on the linter because of "unused record type accessors". Having them starting with a `_` turns off that warning already.